centreon/src/Core/TimePeriod/Infrastructure/Repository/DbReadTimePeriodRepository.php

```diff
@@ -0,0 +1,316 @@
+        $statement = $this->db->prepare(
+            $this->translateDbName('SELECT * FROM `:db`.timeperiod WHERE tp_id = :id')
+        );
+        $statement->bindValue(':id', $timePeriodId, \PDO::PARAM_INT);
+        $statement->execute();
+        if (($result = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            /**
+             * @var array{
+             *     tp_id: int,
+             *     tp_name: string,
+             *     tp_alias: string,
+             *     tp_monday: string,
+             *     tp_tuesday: string,
+             *     tp_wednesday: string,
+             *     tp_thursday: string,
+             *     tp_friday: string,
+             *     tp_saturday: string,
+             *     tp_sunday: string,
+             *     template_id: int
+             * } $result
+             */
+            $newTimePeriod = $this->createTimePeriod($result);
+            $timePeriod[$newTimePeriod->getId()] = $newTimePeriod;
+            $this->addTemplates($timePeriod);
+            $this->addExtraTimePeriods($timePeriod);
+            return $timePeriod[$newTimePeriod->getId()];
+        }
+        return null;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findByRequestParameter(RequestParametersInterface $requestParameters): array
+    {
+        $this->info('Find time periods by request parameter');
+        $sqlRequestTranslator = new SqlRequestParametersTranslator($requestParameters);
+        $sqlRequestTranslator->setConcordanceArray([
+            'id' => 'tp_id',
+            'name' => 'tp_name',
+            'alias' => 'tp_alias',
+        ]);
+        $request = $this->translateDbName('SELECT SQL_CALC_FOUND_ROWS tp.* FROM `centreon`.timeperiod tp');
+
+        // Search
+        $request .= $sqlRequestTranslator->translateSearchParameterToSql();
+
+        // Sort
+        $sortRequest = $sqlRequestTranslator->translateSortParameterToSql();
+        $request .= !is_null($sortRequest)
+            ? $sortRequest
+            : ' ORDER BY tp_id ASC';
+
+        // Pagination
+        $request .= $sqlRequestTranslator->translatePaginationToSql();
+
+        $statement = $this->db->prepare($request);
+        foreach ($sqlRequestTranslator->getSearchValues() as $key => $data) {
+            $type = key($data);
+            if ($type !== null) {
+                $value = $data[$type];
+                $statement->bindValue($key, $value, $type);
+            }
+        }
+        $statement->execute();
+        $result = $this->db->query('SELECT FOUND_ROWS()');
+        if ($result !== false && ($total = $result->fetchColumn()) !== false) {
+            $sqlRequestTranslator->getRequestParameters()->setTotal((int) $total);
+        }
+        /**
+         * @var array<int, TimePeriod> $timePeriods
+         */
+        $timePeriods = [];
+
+        while (($result = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            /**
+             * @var array{
```

Comment:
Up to you, cf. idea of [@phpstan-type](https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases)